pages/api/users/[username]/index.js

```diff
@@ -14,12 +17,16 @@ export default async function handler(req, res) {
       .status(400)
       .json({ error: "Invalid request: GET request required" });
   }
-
-  const { status, profile } = await getUserApi(req.query.username);
+  let sameUser = false;
+  const session = await unstable_getServerSession(req, res, authOptions);
```

Comment:
This needs to happen in the `getUserApi` function, because the UI calls that function directly

Any API calls will also call that function via this `handler` function

Reply:
I think you can pass in `req` and `res` in the file `pages/[username].js` and update `getUserApi` to use them as parameters

Reply:
```diff
diff --git a/pages/[username].js b/pages/[username].js
index 3f26a95b..cf561e03 100644
--- a/pages/[username].js
+++ b/pages/[username].js
@@ -15,14 +15,14 @@ import Page from "@components/Page";
 import UserPage from "@components/user/UserPage";
 
 export async function getServerSideProps(context) {
-  const { req } = context;
+  const { req, res } = context;
   const username = context.query.username;
   const log = logger.child({
     username: username,
     ip: requestIp.getClientIp(req),
   });
 
-  const { status, profile } = await getUserApi(username);
+  const { status, profile } = await getUserApi(req, res, username);
   if (status !== 200) {
     log.error(
       profile.error,
diff --git a/pages/api/users/[username]/index.js b/pages/api/users/[username]/index.js
index 56780ec1..9f088e32 100644
--- a/pages/api/users/[username]/index.js
+++ b/pages/api/users/[username]/index.js
@@ -1,5 +1,7 @@
 import connectMongo from "@config/mongo";
 import logger from "@config/logger";
+import { authOptions } from "../../auth/[...nextauth]";
+import { unstable_getServerSession } from "next-auth/next";
 
 import Profile from "@models/Profile";
 import Stats from "@models/Stats";
@@ -15,11 +17,14 @@ export default async function handler(req, res) {
       .json({ error: "Invalid request: GET request required" });
   }
 
-  const { status, profile } = await getUserApi(req.query.username);
+  const { status, profile } = await getUserApi(req, res, req.query.username);
   return res.status(status).json(profile);
 }
 
-export async function getUserApi(username) {
+export async function getUserApi(req, res, username) {
+  const session = await unstable_getServerSession(req, res, authOptions);
+  console.log(session);
+
   await connectMongo();
 
   const log = logger.child({ username: username });
```

Reply:
Made all the changes. Ready for review